presto-accumulo/src/main/java/com/facebook/presto/accumulo/io/AccumuloPageSink.java

```diff
@@ -0,0 +1,301 @@
+                rowIdOrdinal = ach.getOrdinal();
+                break;
+            }
+        }
+
+        if (rowIdOrdinal == null) {
+            throw new PrestoException(INTERNAL_ERROR, "Row ID ordinal not found");
+        }
+
+        this.serializer = table.getSerializerInstance();
+
+        try {
+            Connector conn = AccumuloClient.getAccumuloConnector(config);
+            // Create a BatchWriter to the Accumulo table
+            BatchWriterConfig conf = new BatchWriterConfig();
+            wrtr = conn.createBatchWriter(table.getFullTableName(), conf);
+
+            // If the table is indexed, create an instance of an Indexer, else null
+            if (table.isIndexed()) {
+                indexer = Optional.of(new Indexer(conn,
+                        conn.securityOperations().getUserAuthorizations(config.getUsername()),
+                        table, conf));
+            }
+            else {
+                indexer = Optional.empty();
+            }
+        }
+        catch (AccumuloException | AccumuloSecurityException e) {
+            throw new PrestoException(UNEXPECTED_ACCUMULO_ERROR,
+                    "Accumulo error when creating BatchWriter and/or Indexer", e);
+        }
+        catch (TableNotFoundException e) {
+            throw new PrestoException(ACCUMULO_TABLE_DNE,
+                    "Accumulo error when creating BatchWriter and/or Indexer, table does not exist", e);
+        }
+    }
+
+    /**
+     * Converts a {@link Row} to an Accumulo mutation.
+     *
+     * @param row Row object
+     * @param rowIdOrdinal Ordinal in the list of columns that is the row ID. This isn't checked at all, so I
+     * hope you're right. Also, it is expected that the list of column handles is sorted
+     * in ordinal order. This is a very demanding function.
+     * @param columns All column handles for the Row, sorted by ordinal.
+     * @param serializer Instance of {@link AccumuloRowSerializer} used to encode the values of the row to
+     * the Mutation
+     * @return Mutation
+     */
+    public static Mutation toMutation(Row row, int rowIdOrdinal, List<AccumuloColumnHandle> columns,
+            AccumuloRowSerializer serializer)
+    {
+        // Set our value to the row ID
+        Text value = new Text();
+        Field rField = row.getField(rowIdOrdinal);
+        if (rField.isNull()) {
+            throw new PrestoException(VALIDATION, "Column mapped as the Accumulo row ID cannot be null");
+        }
+
+        setText(rField, value, serializer);
+
+        // Iterate through all the column handles, setting the Mutation's columns
+        Mutation m = new Mutation(value);
+
+        // Store row ID in a special column
+        m.put(ROW_ID_COLUMN, ROW_ID_COLUMN, new Value(value.copyBytes()));
+
+        for (AccumuloColumnHandle ach : columns) {
+            // Skip the row ID ordinal
+            if (ach.getOrdinal() == rowIdOrdinal) {
```

Comment:
Since `rowIdOrdinal` is `Integer` you probably should use `equals(Object)` instead. Or change `rowIdOrdinal` to primitive `int` type.